### DISCUSSION TITLE
how to let rand() return different values in a row?

### DISCUSSION THREAD

**Discussion by l1t1** [ClickHouse/ClickHouse#63012] [General]
now it returns the same value
```
SELECT
    rand(),
    rand()

Query id: c1d817bc-b23d-4e67-bb3c-838515ae613d

   ┌─────rand()─┬─────rand()─┐
1. │ 1879788443 │ 1879788443 │
   └────────────┴────────────┘
```

**Comment by l1t1** [comment]
duckdb returns different values
```
D select random(),random();
┌────────────────────┬────────────────────┐
│      random()      │      random()      │
│       double       │       double       │
├────────────────────┼────────────────────┤
│ 0.8796781469136477 │ 0.7237269899342209 │
└────────────────────┴────────────────────┘
```

**Comment by Algunenano** [comment]
https://clickhouse.com/docs/en/sql-reference/functions/random-functions

> All functions in this section accept zero or one arguments. The only use of the argument (if provided) is to prevent [common subexpression elimination](https://clickhouse.com/docs/en/sql-reference/functions#common-subexpression-elimination) such that two different executions within a row of the same random function return different random values.

```
SELECT
    rand(1),
    rand(2)

Query id: 224d8542-fb6d-447d-972c-e76731d3e559

   ┌────rand(1)─┬────rand(2)─┐
1. │ 2139910815 │ 1154787474 │
   └────────────┴────────────┘
```

Otherwise CH omits one of the calls.
